### PR TITLE
test: add migrate db seed test

### LINKED ISSUE #1806: Test seeding
Run steps from https://github.com/prisma/prisma/issues/7176 in CI here and make sure data is available afterwards.

### PR DESCRIPTION
**Pull request by Jolg42**
Based on and replaces https://github.com/prisma/e2e-tests/pull/1807

closes #1806

@janpio this adds 12 jobs 😨

### CODE REVIEW

**Comment by Jolg42** [comment]
I think we should remove the enginetype from the matrix here, it doesn’t bring any value I can see (it's tested in other tests and that would reduce the number of jobs here)

**Review by millsp — APPROVED** [review]

File: migrate/db-seed-commonjs-pkg/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "prisma": "2.31.0-dev.46",
+    "ts-node": "10.2.1",
+    "jest": "26.6.3",
+    "prettier": "2.3.1"
+  },
+  "dependencies": {
+    "@prisma/client": "2.31.0-dev.46"
+  },
+  "scripts": {
+    "test": "jest --forceExit"
```
Comment (janpio) [inline migrate/db-seed-commonjs-pkg/package.json]: Why `forceExit`? I think we should Jest let us tell about stuff that stays open:
```suggestion
    "test": "jest"
```

File: migrate/db-seed-commonjs-pkg/prisma/seed.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import { PrismaClient, Prisma } from '@prisma/client'
+
+const prisma = new PrismaClient()
+
+const userData: Prisma.UserCreateInput[] = [
```
Comment (janpio) [inline migrate/db-seed-commonjs-pkg/prisma/seed.ts:5]: Is this best practice how to define input for our create methods? 💡

File: migrate/db-seed-commonjs-pkg/test.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+set -eux
+
+yarn prisma db pull --print
```
Comment (janpio) [inline migrate/db-seed-commonjs-pkg/test.sh:5]: confirmation that `db push` worked?

File: migrate/db-seed-commonjs-pkg/run.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+set -eux
+
+yarn install
+
+# generate not needed, it will be done by db push
+# yarn prisma generate
+yarn prisma db push --force-reset
```
Comment (janpio) [inline migrate/db-seed-commonjs-pkg/run.sh:9]: if `force-reset` does not work for whatever reason, the test would possibly also succeed if the seeding fails, correct?

File: migrate/db-seed-esm-pkg/package.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    "prisma": "2.31.0-dev.46",
+    "ts-node": "10.2.1",
+    "jest": "26.6.3",
+    "prettier": "2.3.1"
+  },
+  "dependencies": {
+    "@prisma/client": "2.31.0-dev.46"
+  },
+  "scripts": {
+    "test": "jest --forceExit"
```
Comment (janpio) [inline migrate/db-seed-esm-pkg/package.json]:
```suggestion
    "test": "jest"
```

**Review by janpio — APPROVED** [review]

File: migrate/db-seed-commonjs-pkg/test.sh (hunk 3 above)
Comment (Jolg42) [reply migrate/db-seed-commonjs-pkg/test.sh:5]: It's a copy paste from your script, so probably

File: migrate/db-seed-commonjs-pkg/prisma/seed.ts (hunk 2 above)
Comment (Jolg42) [reply migrate/db-seed-commonjs-pkg/prisma/seed.ts:5]: If you want to extract the object outside the prisma call, yes

File: migrate/db-seed-commonjs-pkg/run.sh (hunk 4 above)
Comment (Jolg42) [reply migrate/db-seed-commonjs-pkg/run.sh:9]: Well here it's about preparing the db and I assume test.sh runs after with
```
#!/bin/sh

set -eux

yarn prisma db pull --print

yarn prisma db seed

yarn test
```

This should fail

**Pull request merged**